be/src/exprs/expr.h

```diff
@@ -229,24 +221,28 @@ class Expr {
     virtual StatusOr<LLVMDatum> generate_ir_impl(ExprContext* context, JITContext* jit_ctx);
 
     // Return true if this expression supports JIT compilation.
-    virtual bool is_compilable() const { return false; }
+    virtual bool is_compilable(RuntimeState* state) const { return false; }
 
-    std::string jit_func_name() const;
+    std::string jit_func_name(RuntimeState* state) const;
 
-    virtual std::string jit_func_name_impl() const;
+    virtual std::string jit_func_name_impl(RuntimeState* state) const;
+
+    std::string jit_func_name() const;
 
     // This function will collect all uncompiled expressions in this expression tree.
     // The uncompiled expressions are those expressions which are not supported by JIT, it will become the input of JIT function.
-    void get_uncompilable_exprs(std::vector<Expr*>& exprs);
+    void get_uncompilable_exprs(std::vector<Expr*>& exprs, RuntimeState* state);
 
     // This method attempts to traverse the entire expression tree from the current expression downwards, seeking to replace expressions with JITExprs.
     // This method searches from top to bottom for compilable expressions.
     // Once a compilable expression is found, it skips over its compilable subexpressions and continues the search downwards.
     // TODO(Yueyang): The algorithm is imperfect and may further be optimized in the future.
-    Status replace_compilable_exprs(Expr** expr, ObjectPool* pool, bool& replaced);
+    Status replace_compilable_exprs(Expr** expr, ObjectPool* pool, RuntimeState* state, bool& replaced);
 
     // Establishes whether the current expression should undergo compilation.
-    bool should_compile() const;
+    bool should_compile(RuntimeState* state) const;
+
+    virtual JitScore compute_jit_score(RuntimeState* state) const;
```

Comment:
better to write some comments describing how each operator calculates the score.